pkg/engine/planner/logical/planner_test.go

```diff
@@ -257,3 +257,355 @@ func TestCanExecuteQuery(t *testing.T) {
 		})
 	}
 }
+
+func TestConvertStringHintsFunction(t *testing.T) {
+	t.Run("converts int64 hint", func(t *testing.T) {
+		stringHints := map[string]string{
+			"bytes": TypeHintInt64,
+		}
+
+		result := convertStringHintsToNumericType(stringHints)
+
+		require.NotNil(t, result)
+		require.Contains(t, result, "bytes")
+		require.Equal(t, NumericInt64, result["bytes"])
+	})
+
+	t.Run("converts float64 hint", func(t *testing.T) {
+		stringHints := map[string]string{
+			"response_time": TypeHintFloat64,
+		}
+
+		result := convertStringHintsToNumericType(stringHints)
+
+		require.NotNil(t, result)
+		require.Contains(t, result, "response_time")
+		require.Equal(t, NumericFloat64, result["response_time"])
+	})
+
+	t.Run("converts duration hint to float64", func(t *testing.T) {
+		stringHints := map[string]string{
+			"duration_field": TypeHintDuration,
+		}
+
+		result := convertStringHintsToNumericType(stringHints)
+
+		require.NotNil(t, result)
+		require.Contains(t, result, "duration_field")
+		require.Equal(t, NumericFloat64, result["duration_field"])
+	})
+
+	t.Run("handles multiple hints", func(t *testing.T) {
+		stringHints := map[string]string{
+			"bytes":         TypeHintInt64,
+			"response_time": TypeHintFloat64,
+			"duration":      TypeHintDuration,
+		}
+
+		result := convertStringHintsToNumericType(stringHints)
+
+		require.NotNil(t, result)
+		require.Len(t, result, 3)
+		require.Equal(t, NumericInt64, result["bytes"])
+		require.Equal(t, NumericFloat64, result["response_time"])
+		require.Equal(t, NumericFloat64, result["duration"])
+	})
+
+	t.Run("returns nil for nil input", func(t *testing.T) {
+		result := convertStringHintsToNumericType(nil)
+
+		require.Nil(t, result)
+	})
+
+	t.Run("returns empty map for empty input", func(t *testing.T) {
+		stringHints := map[string]string{}
+
+		result := convertStringHintsToNumericType(stringHints)
+
+		require.NotNil(t, result)
+		require.Empty(t, result)
+	})
+}
+
+func TestPlannerCreatesParseFromLogfmt(t *testing.T) {
```

Comment:
Can you update all of these so they assert against the generated SSA like the other planner tests, please?